### DISCUSSION TITLE
JsonField  source and validate(self, attrs) might have a mismatch

### DISCUSSION THREAD

**Discussion by kxtells** [encode/django-rest-framework#9478] [Potential Issue]
Hello everyone.

I have a situation that seemed strange at first and thought to ask if it is intended behaviour or something unexpected for 3.15.1.

In this code example I provide the smallest example I could come up with that creates this situation.

In this case I have a Comment class, that I serialize via a `CommentSerializer` that declares a specific `source` function to fill up the given field.

Alongside this we have some specific `validate(self, data)` for this serializer.

I would expect the validation payload to come as
```
{'content': 'aha'}
```

But instead it comes as
```
{'myFunc': 'aha'}
```

I imagine that if the source points to an attribute this behaviour would not be too bad, but when pointing to a function it seems strange to me.

As said, I'm just wondering if this is intended or it's some unexpected behaviour.

```python
from rest_framework import serializers

class Comment():
    def __init__(self, txt):
        self.content = txt

    def myFunc(self):
        return "POTATO"

class CommentSerializer(serializers.Serializer):
    content = serializers.CharField(max_length=200, source="myFunc")

    def validate(self, data):
        print(data)
        return data

comment = Comment("HelloDear")
serializer = CommentSerializer(data={"content": "aha"})
print(serializer.is_valid())
```

Salut, and thanks for the hard work!

**Comment by sevdog** [comment]
For what I know this is a behaviour which has been in DRF for long time.

Maybe it is not well documented, the `source` attribute of fields works in both read **and** in write operations. The meaning of this is to have an "internal" name (the source) and an "external" name (field name) for the same field, in this way you may change your internal data structure without having the API users to change too.
